### PR TITLE
[Merged by Bors] - feat: Lattice closedness

### PR DESCRIPTION
**Pull request by YaelDillies**
Define when a set in a lattice is closed under lattice operations.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Closure.lean [hunk 1]
```diff
@@ -272,7 +271,25 @@ end SemilatticeSup
 
 section CompleteLattice
 
-variable [CompleteLattice α] (c : ClosureOperator α)
+variable [CompleteLattice α] (c : ClosureOperator α) {p : α → Prop}
+
+/-- Define a closure operator from a predicate that's preserved under infima. -/
+def ofPred (p : α → Prop) (hsinf : ∀ s, (∀ a ∈ s, p a) → p (sInf s)) : ClosureOperator α :=
+  ClosureOperator.mk₃ (λ a ↦ ⨅ b : {b // p b ∧ a ≤ b}, b) p
```
Comment (eric-wieser) [inline Mathlib/Order/Closure.lean:278]: `fun`

File: Mathlib/Order/SupClosed.lean [hunk 2]
```diff
@@ -123,22 +160,69 @@ end SemilatticeInf
 open Finset OrderDual
 
 section Lattice
-variable [Lattice α]
+variable {ι : Sort*} [Lattice α] [Lattice β] {S : Set (Set α)} {f : ι → Set α} {s t : Set α} {a : α}
+
+open Set
+
+/-- A set `s` is *lattice-closed* if `a ⊔ b ∈ s` and `a ⊓ b ∈ s` for all `a ∈ s`, `b ∈ s`. -/
```
Comment (jcommelin) [inline Mathlib/Order/SupClosed.lean:167]: Is this also the name used in the literature? Or does this concept not exist in the mainstream literature?

Comment (YaelDillies) [reply Mathlib/Order/SupClosed.lean:167]: This concept definitely exists in the literature: it's caled a sublattice. However I am trying to reserve this name for the bundled object, which I'm defining in #7549.

What do you think? Should I also say this is the property of being a sublattice? in the docstring or in the name?

Comment (jcommelin) [reply Mathlib/Order/SupClosed.lean:167]: I would definitely mention this in the docstring. And I would even consider calling this `isSublattice` or something like that.

Comment (eric-wieser) [reply Mathlib/Order/SupClosed.lean:167]: I guess the argument for the current name is consistency with `SupClosed` and `InfClosed`.

Comment (YaelDillies) [reply Mathlib/Order/SupClosed.lean]: I've renamed it to `IsSublattice` for now.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**